bin/doctrine-dbal.php

```diff
@@ -1,57 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-use Doctrine\DBAL\Tools\Console\ConsoleRunner;
-
-fwrite(
-    STDERR,
-    '[Warning] The use of this script is discouraged.'
-        . ' You find instructions on how to boostrap the console runner in our documentation.'
-        . PHP_EOL
-);
-
-echo PHP_EOL . PHP_EOL;
-
-$files       = [__DIR__ . '/../vendor/autoload.php', __DIR__ . '/../../../autoload.php'];
-$loader      = null;
-$cwd         = getcwd();
-$directories = [$cwd, $cwd . DIRECTORY_SEPARATOR . 'config'];
-$configFile  = null;
-
-foreach ($files as $file) {
-    if (file_exists($file)) {
-        $loader = require $file;
-
-        break;
-    }
-}
-
-if (! $loader) {
-    throw new RuntimeException('vendor/autoload.php could not be found. Did you run `php composer.phar install`?');
-}
-
-foreach ($directories as $directory) {
-    $configFile = $directory . DIRECTORY_SEPARATOR . 'cli-config.php';
-
-    if (file_exists($configFile)) {
-        break;
-    }
-}
-
-if (! file_exists($configFile)) {
-    ConsoleRunner::printCliConfigTemplate();
-
-    exit(1);
-}
-
-if (! is_readable($configFile)) {
-    echo 'Configuration file [' . $configFile . '] does not have read permission.' . PHP_EOL;
-
-    exit(1);
-}
-
-$commands           = [];
-$connectionProvider = require $configFile;
-
-ConsoleRunner::run($connectionProvider, $commands);
```

Comment:
@derrabus so now that this file is removed, what's the role of the `ConsoleRunner` class? I see that it still exists but it's not used in the code. It doesn't seem to be used in DoctrineBundle either.

From the details you posted earlier in https://github.com/doctrine/dbal/issues/5076#issuecomment-985050523

> [...] the DBAL and ORM commands are integrated into the applications console by DoctrineBundle. 

It doesn't look like the runner is used there. It seems to have been used only by this file which no longer exists.

Should `ConsoleRunner` be deprecated and removed entirely?

Reply:
> In my understanding, it's a boilerplate code used to bootstrap a Symfony Console application.

Yes.

> It may be useful only if the application that depends on the DBAL doesn't bootstrap one itself.

No. If your application bootstraps it's own console application, you can call `ConsoleRunner::addCommands()` to add the DBAL's commands to it. If it doesn't, `ConsoleRunner::run()` bootstraps the application for you and runs it.

> Why should the DBAL maintain the code that should be possible to copy-paste from the documentation?

The idea is that `ConsoleRunner::run()` will remain a stable piece of boilerplate, even if we add/remove/rename command classes. Removing the class would complicate bootstrapping the console application unnecessarily.

> But it's not tested, so it's still mostly dead code.

Let's add tests then.

Reply:
> No. If your application bootstraps it's own console application, you can call `ConsoleRunner::addCommands()` to add the DBAL's commands to it.

There's currently just one command (the other is deprecated). `ConsoleRunner::addCommands()` accepts the same parameters as `Application::addCommands()`, i.e. requires a `ConnectionProvider` to be passed. What value does it provide this way?

> The idea is that ConsoleRunner::run() will remain a stable piece of boilerplate [...]

That's what the documentation for the CLI framework is for. As soon as a user wants to make a change in their CLI application behavior (e.g. change the name, implement their own error handling), they will have to copy-paste our code. It's better to have them do that from the beginning.

Reply:
> The current code violates the SRP

I really fail to see why this is a problem for bootstrap code, tbh.

I'm sorry, but assumed we've settled the discussion around the CLI tools already. The amount of time and energy I can invest in open source work these days is limited, so I'd rather not spend it on defending a piece of boostrap logic that I haven't even written myself.

Please remove `ConsoleRunner` if it really bugs you that much.